### ISSUE TITLE
Accessibility: Issues with Screen Readers on `domain/manage` Page

### ISSUE THREAD

**Issue by karenroldan** [Automattic/wp-calypso#79413]
### Quick summary

**Context:** pdtkmj-1Cx-p2#comment-2952

> On domains/manage, I can’t get a screen reader to read the status or expiration date of the domains. ‘Auto-renew’ and ‘Actions’ headers are set to tabindex -1 and won’t be read by a screen reader in the same sequence as the rest of the table headers.

> The domains/manage page has TWO ‘main’ landmarks. It should only have one.

> The auto-renew toggle button needs to have an accessible name or label. It should also report the status in a way that will be read by a screen reader.

### Steps to reproduce

1. Start at `/domains/manage`
2. Turn on the screen reader. For example, if you are on Mac, you could use VoiceOver.
3. Navigate through the elements on the page.
4. Check if the screen reader can read the domain's status and expiration date.
5. Check if the screen reader can read the ‘Auto-renew’ and ‘Actions’ headers in the same sequence. 
6. Finally, navigate to the auto-renew toggle button. Check if it has an accessible name or label that can be ready by the screen reader. Try to toggle it on and off, and listen to see if the screen reader reports the status.

### What you expected to happen

The screen reader can read the elements on the page following the same sequence. 

### What actually happened

The screen reader cannot read the following:
- the status or expiration of the domains
- `Auto-renew` and `Actions` headers
- `Auto-renew` toggle button and its status. 

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by worldomonation** [comment]
📌 TESTED AND VERIFIED

Revision: 0b675812d798f3d95487b121fdcb2894e09a96c2

- using VoiceOver, Tab through the elements on page in `/domains/manage`.
- VoiceOver could not read the Status and Registered Until columns.
- related, VoiceOver also did not read the "Auto-renew" column.

https://github.com/Automattic/wp-calypso/assets/6549265/6ce7a90c-2983-4e31-a1b2-b88f15997943

**Comment by ciampo** [comment]
👋  Not sure if there's a better place to leave initial feedback, but having each interactive element in the table be a tab stop is not a great experience for the user.
Imagine a keyboard user who needs to focus an item on the 10th—the user may need to press tab 30+ times before reaching their target!

A better solution would be to make each row a [composite widget](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/composite_role), or even make the whole table a grid composite widget.
